### PR TITLE
PANOS Sensor For Active GPVPN Tunnels

### PR DESCRIPTION
**Pull request by eg2965**
Added a PANOS sensor to monitor active GlobalProtect VPN tunnels (aka sessions) as a way to track VPN availability on a PA firewall.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by eg2965** [comment]
Adding screenshots of my Dev instance

PANOS-GPVPN-Settings-Health-Tab

PANOS-GPVPN-Settings-Health-Tab

PANOS-GPVPN-Yaml

**Comment by eg2965** [comment]
@laf @murrant - I can't seem to get past the mysql and mariadb unit tests. I've tried several test json files, however each time they fail with the message "OS panos: Discovered sensors data does not match that found in tests/data/panos_vsys.json". I've checked that file, and it looks like old PA-3020 testing data. I tried manually copying my GlobalProtect sensor data to the panos_vsys.json file but it stills fails. Unfortunately the error message is not giving me enough information to find the root cause of the issue. Can you assist?

**Comment by PipoCanaja** [comment]
Hi @eg2965 
For the existing JSON, you need to run only save-test-data.php against them, if they already have GlobalProtect data in the matching SNMPREC. If not, there is absolutely nothing to change.  
And you also need to create a new SNMPREC file for your variant (which will include the GlobalProtect data for sure) and run  save-test-data.php to create the matching JSON.

**Comment by eg2965** [comment]
Thanks @PipoCanaja - that was the piece I was missing. I just re-ran the ./scripts/save-test-data.php command with modifiers of -o panos -v 'vsys', and uploaded the modified panos_vsys.json file here. The checks passed! Will submit the pull request for review now.

**Comment by eg2965** [comment]
Clearly I submitted way more commits than I would've like to on account of the failing tests. If you guys would prefer me to delete this PR/branch and start fresh, I'd be willing to. Let me know

**Comment by PipoCanaja** [comment]
> Clearly I submitted way more commits than I would've like to on account of the failing tests. If you guys would prefer me to delete this PR/branch and start fresh, I'd be willing to. Let me know

If you have the result you wanted, we don't care about the amount of commits to reach this state. We'll squash the PR to one commit anyway to merge it so don't worry.

File: tests/snmpsim/panos.snmprec
Comment (PipoCanaja) [inline tests/snmpsim/panos.snmprec:1]: This file as well as the matching JSON are now missing. Could you please restore this one back to its original state, and run save-test-data.php against it to restore je JSON ?

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

File: tests/snmpsim/panos.snmprec
Comment (eg2965) [reply tests/snmpsim/panos.snmprec:1]: @PipoCanaja - I took the long road, but finally got there. Everything should be in order now.
Let me know if there's anything else you need before merging.

**Review by laf — APPROVED** [review]
LGTM, thanks for the contribution @eg2965

**Pull request merged**